packages/teams-js/src/public/presence.ts

```diff
@@ -0,0 +1,226 @@
+   */
+  upn: string;
+}
+
+/**
+ * Parameters for setting presence
+ */
+export interface SetPresenceParams {
+  /**
+   * New presence status to set
+   */
+  status: PresenceStatus;
+
+  /**
+   * Optional custom status message
+   * @since 2.0.0
+   */
+  customMessage?: string;
+}
+
+/**
+ * Response handler for presence information
+ */
+class UserPresenceResponseHandler extends ResponseHandler<UserPresence, UserPresence> {
+  public validate(response: UserPresence): boolean {
+    return response !== undefined && Object.values(PresenceStatus).includes(response.status);
+  }
+
+  public deserialize(response: UserPresence): UserPresence {
+    return response;
+  }
+}
+
+/**
+ * Response handler for void responses
+ */
+class VoidResponseHandler extends ResponseHandler<void, void> {
+  public validate(response: void): boolean {
+    return response === undefined;
+  }
+
+  public deserialize(response: void): void {
+    return response;
+  }
+}
+
+/**
+ * Serializable wrapper for GetPresenceParams
+ */
+class SerializableGetPresenceParams implements ISerializable {
+  public constructor(private params: GetPresenceParams) {}
+
+  public serialize(): object {
+    return this.params;
+  }
+}
+
+/**
+ * Serializable wrapper for SetPresenceParams
+ */
+class SerializableSetPresenceParams implements ISerializable {
+  public constructor(private params: SetPresenceParams) {}
+
+  public serialize(): object {
+    return this.params;
+  }
+}
+
+/**
+ * Gets a user's current presence status
+ * @param params Parameters for getting presence
+ * @returns Promise resolving with the user's presence
+ *
+ * @throws Error if:
+ * - The presence capability is not supported
+ * - The library has not been initialized
+ * - The UPN parameter is invalid
+ */
+export function getPresence(params: GetPresenceParams): Promise<UserPresence> {
+  ensureInitialized(runtime, FrameContexts.content);
+
+  if (!isSupported()) {
+    throw errorNotSupportedOnPlatform;
+  }
+
+  validateUpn(params.upn);
+
+  return callFunctionInHostAndHandleResponse(
+    'presence.getPresence',
+    [new SerializableGetPresenceParams(params)],
+    new UserPresenceResponseHandler(),
+    getApiVersionTag(presenceTelemetryVersionNumber, ApiName.Presence_GetPresence),
+  );
+}
+
+/**
+ * Sets the current user's presence status
+ * @param params Parameters for setting presence
+ * @returns Promise that resolves when operation completes
+ *
+ * @throws Error if:
+ * - The presence capability is not supported
+ * - The library has not been initialized
+ * - The status parameter is invalid
+ */
+export function setPresence(params: SetPresenceParams): Promise<void> {
+  ensureInitialized(runtime, FrameContexts.content);
+
+  if (!isSupported()) {
+    throw errorNotSupportedOnPlatform;
+  }
+
+  validateStatus(params.status);
+
+  return callFunctionInHostAndHandleResponse(
+    'presence.setPresence',
+    [new SerializableSetPresenceParams(params)],
+    new VoidResponseHandler(),
```

Comment:
You actually can use the simpler version of this since you don't expect to get anything back except an SdkError in the case that the host has an error it wants to return to the user. Check out `callFunctionInHost`. You can see some examples of it being used in `externalAppAuthenticationForCEA.ts`